x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/EqlPlugin.java

```diff
@@ -60,6 +60,20 @@ public class EqlPlugin extends Plugin implements ActionPlugin, CircuitBreakerPlu
         Setting.Property.DeprecatedWarning
     );
 
+    public static final Setting<Boolean> DEFAULT_ALLOW_PARTIAL_SEARCH_RESULTS = Setting.boolSetting(
+        "xpack.eql.default_allow_partial_results",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    public static final Setting<Boolean> DEFAULT_ALLOW_PARTIAL_SEQUENCE_RESULTS = Setting.boolSetting(
+        "xpack.eql.default_allow_partial_sequence_results",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
```

Comment:
Do we need a setting to tweak the default given it's a flag exposed at the request level?

Reply:
I'm not completely sure tbh (maybe @yctercero has opinions here).
My thinking is that it could make our life easier on existing deployments, where we have hundreds of EQL rules already configured and we don't want to change them one by one, but maybe this can be addressed at higher level (in Kibana?).

Reply:
Yes, all the defaults are to `false` now.
The discussion here was more about the real need for a cluster setting to change these defaults, since we can always set these values per query.

Reply:
I'm with @javanna here. If there is no need for this, let's not make it more complicated than it should be. If there will be a desire in the future for this, we can add it.

The basic principle of **"it's easy to put stuff in, harder to take it out"** (think about BWC) applies here.

Same principle for having this as both **request parameter** and **request body parameter**: it complicates the logic unnecessarily and confuses users with a lot of options to think and reason about. I realize that ES query DSL has this in the same way (both **request parameter** and **request body parameter**) and I am not against this that much, but unless there is an actual need and request I wouldn't confuse users.

Reply:
Let's go with the less complicated option where we would handle it on the Kibana side - say an advanced setting or something of the like. 

cc @approksiu